docs/go/core/bzlmod.md

```diff
@@ -160,6 +160,24 @@ go_deps.module(
 )
 ```
 
+#### Specifying Workspaces
+
+`go.work` is supported by the go_deps module extension provided by Gazelle.
```

Comment:
Should we mention any restrictions, e.g. the version conflict handling and the fact that go.work is only supported in the root module?

Reply:
@fmeum I've taken a stab \w a limitations sections, let me know what you think!